### DISCUSSION TITLE
Export data to D3

### DISCUSSION THREAD

**Discussion by dimm0** [vesoft-inc/nebula#4216] [Q&A]
**General Question**

Hi! Is there any connection from free nebula to D3 (observableHQ) available? How can I export the data from the database and show it in D3? The examples in blog assume the data is already there

Thanks!

**Comment by jamieliu1023** [comment]
@nianiaJR @NicolaCage Could you please help look into this?

**Comment by jerryliang64** [comment]
@dimm0 We have a free project (https://github.com/vesoft-inc/nebula-studio) to help nebula user use graph database by GUI, which include the data show via d3.js.It's an open source project that you can refer.🤝

**Comment by dimm0** [comment]
I am already running nebula in kubernetes by operator and studio by helm chart, but I don't see the D3 viz in my studio:

<img width="851" alt="Screen Shot 2022-04-28 at 9 56 30 PM" src="https://user-images.githubusercontent.com/1425007/165887222-d39246db-96ba-419b-9c7e-430b76e0b51f.png">

Is it a different studio without a helm chart?

**Comment by dimm0** [comment]
Also I already have my own D3 project (https://traceroute.nrp-nautilus.io) which pulls data from our cluster via the networkx python package - it's json format can be directly imported to D3. I was wondering if I can somehow do the same with Nebula.

**Reply by jerryliang64** [reply]
1. You can click the `Console` tab，when you do a subgraph query and then you can find a selection to show the result by visualization, which is done by D3.
2. I'm a little confused about your question: "if I can some how do the same with Nebula"?  Nebula is a graph database to store your data like nodes and edges that you show in the d3 network demo.So I guess that you want to know how to query these data from nebula and then to use theme in D3 visualization?

**Reply by jerryliang64** [reply]
If you wang to do the  same with Nebula. Here are the steps you can refer:

1. build the schema to fit in the data that you show in the network.
2. import data to the schema
3. Find the proper queries to fetch these data, you can refer our nebula documents: https://docs.nebula-graph.io/3.1.0/
All the steps: schema -> import -> query, you can do it in the nebula studio. 🤝

**Reply by jamieliu1023** [reply]
@nianiaJR I think what @dimm0 meant by "do the same with Nebula" was that if his own D3 project can pull data from Nebula Graph via the networkx python package the same way as he does with his other clusters for visualization. Please correct me if I am wrong.

**Comment by dimm0** [comment]
I need some way to pull data from nebula and import into D3. Networkx is just one example, can be anything (I'd prefer directly querying the nebula API from D3)

Got it, so I need a subgraph query to see something in the studio. It can't just show the whole graph. Will work on that, thanks! :)

**Reply by wey-gu** [reply]
@dimm0 yes, you could fetch vertices and edges and serialize them to be D3 consumable(but not limit to `get subgraph` queries). I have a reference project here https://github.com/wey-gu/nebula-corp-rel-search using a vue2 d3js binding and python backend.

For talking to the nebula, it's normally not be done directly in the browser, instead of from a backend in languages you like. In my case it was python.

**Reply by dimm0** [reply]
Cool thanks!
